cms/core/tests/test_views.py

```diff
@@ -17,3 +19,16 @@ def test_csrf_token_mismatch_logs_an_error(self):
             self.client.post("/admin/login/", {})
 
         self.assertIn("CSRF Failure: CSRF cookie", logs.output[0])
+
+
+class TestCookiesBannerTestCase(TestCase):
+    """Tests for the cookie banner functionality."""
+
+    def test_google_tag_manager_script_present_when_tracking_consent_given(self):
+        """Check that the Google Tag Manager script is present on the page when the user agrees to tracking."""
+        self.client.cookies["ons_cookie_policy"] = {"usage": True}
+
+        response = self.client.get("/")
+
+        self.assertIn("https://www.googletagmanager.com/gtm.js?id=", response.rendered_content)
+        self.assertIn(settings.GOOGLE_TAG_MANAGER_CONTAINER_ID, response.rendered_content)
```

Comment:
Actually, this approach results in a false positive (this will always pass regardless of the cookie), as the GTM script is created via JS, which integration tests don’t execute. Therefore, we can’t reliably verify the presence of the GTM ID here. I recommend removing this test for now, as we can add a functional test using Playwright to reliably test this.

Reply:
After conversation with @MebinAbraham we've decided to keep the test and change its purpose. 
Rather than an integration test checking the GTM/banner functionality, it'll now be a unit test doing a simple check that the GTM ID and URL are present on the page.
The functional tests for GTM/banner will have to follow once we get them set up. These will check the logic for cookie selection and whether the GTM script is present on the page.